pkg/controller/controller_utils.go

```diff
@@ -542,6 +544,39 @@ func GetPodFromTemplate(template *v1.PodTemplateSpec, parentObject runtime.Objec
 	return pod, nil
 }
 
+func (r RealPodControl) validateNodeConflict(pod *v1.Pod) error {
+	if len(pod.Spec.NodeName) == 0 {
+		return nil
+	}
+
+	node, err := r.KubeClient.Core().Nodes().Get(pod.Spec.NodeName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	nodeInfo := schedulercache.NewNodeInfo()
+	err = nodeInfo.SetNode(node)
+	if err != nil {
+		return err
+	}
+
+	fit, reasons, err := predicates.PodSelectorMatches(pod, nil, nodeInfo)
```

Comment:
@kargakis , for this case, it's kubelet who rejects the pods; the scheduler is not involved as the nodeName was assigned by yml.
The scheduler only considers the Pod that nodeName is empty (unassigned) and non-terminated.

```
...
          "spec": {
              "nodeName": "192.168.1.220",
...
```